File: tag_md_maker.py
import re

readPath = "/Users/dongmyeonglee/Procjects/vuepress-blog/docs/.vuepress/store/tag.js"
writePath = "/Users/dongmyeonglee/Procjects/vuepress-blog/docs/tag"

readlines = list(filter(lambda s: s.__contains__('tagStore.put'), open(readPath, 'r').readlines()))


def templte(_tagName):
    return f'---\n' \
           f'sidebar: false\n' \
           f'sitemap:\n' \
           f'   exclude: true\n' \
           f'---\n' \
           f'<TagMain tagName="{_tagName}"/>'


for line in readlines:
    tagWithRest = re.search('\'[A-Za-z.]+\'', line).group(0)
    tagName = tagWithRest[1:len(tagWithRest) - 1]
    writer = open(f'{writePath}/{tagName}.md', 'w')
    writer.write(templte(tagName))

File: tagMDMaker.js
import {getTagStore} from "./docs/.vuepress/store/tag";
import fs from "fs";

const tags = getTagStore().values();

function getTemplate(tagName) {
    const lines = [
        '---',
        'sidebar: false',
        'sitemap:',
        '   exclude: true',
        '---',
        `<TagMain tagName=\"${tagName}\" />`,
    ]
    return lines.reduce((line, curLine) => line + '\n' + curLine)
}

function getWritePath(tagName) {
    return `./docs/tag/${tagName}.md`;
}

tags.forEach(tag => {
    const tagName = tag.tagName;
    const template = getTemplate(tagName);
    const writePath = getWritePath(tagName);
    fs.writeFile(writePath, template, 'utf8', (err => {
        if (err) {
            console.log(err);
            process.exit();
        }
    }))
    console.log('tag create sucess: ', writePath)
})
